src/image/ExposureInfo.cc

```diff
@@ -254,7 +254,10 @@ void ExposureInfo::_readFits(
         }
         int wcsId = popInt(*metadata, "WCS_ID");
         try {
-            _wcs = archive.get<Wcs>(wcsId);
+            auto archiveWcs = archive.get<Wcs>(wcsId);
+            if (archiveWcs) {
+                _wcs = archiveWcs;
+            }
```

Comment:
I fixed the comment

I also note that we could check for wcsId == 0, but doing that would break the symmetry with e.g. Psf handing and I didn't want to fix all of them on this ticket.